core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -872,6 +872,25 @@ class DAGScheduler(
     // will be posted, which should always come after a corresponding SparkListenerStageSubmitted
     // event.
     stage.latestInfo = StageInfo.fromStage(stage, Some(partitionsToCompute.size))
+    val taskIdToLocations = try {
+      stage match {
+        case s: ShuffleMapStage =>
+          partitionsToCompute.map { id => (id, getPreferredLocs(stage.rdd, id))}.toMap
+        case s: ResultStage =>
+          val job = s.resultOfJob.get
+          partitionsToCompute.map { id =>
+            val p: Int = job.partitions(id)
+            (id, getPreferredLocs(stage.rdd, p))
+          }.toMap
+      }
+    } catch {
+      case NonFatal(e) =>
```

Comment:
Because `getPreferredLocs()` may throw exception, please refer to (https://github.com/apache/spark/commit/0b5abbf5f96a5f6bfd15a65e8788cf3fa96fe54c).